src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -160,9 +166,12 @@ InboundLedger::init(ScopedLockType& collectionLock)
 std::size_t
 InboundLedger::getPeerCount() const
 {
-    return std::count_if(mPeers.begin(), mPeers.end(), [this](auto id) {
-        return app_.overlay().findPeerByShortID(id) != nullptr;
+    std::size_t count = 0;
+    mPeerSet->visitAddedPeers([this, &count](auto id) {
+        if (app_.overlay().findPeerByShortID(id) != nullptr)
+            ++count;
     });
+    return count;
```

Comment:
I am not sure if PeerSet should be a container. In addition, the current container kind of use cases probably should be avoid if possible, maybe difficult. I feel the coupling among different components are a bit tight. 

If we return a `const&` of the privately-held container, then we lose the lock protecting the container. We can return a copy. Or we can remove the lock, make the `PeerSet` not thread-safe, and ask the users of `PeerSet` to provide thread-safety.

Reply:
Right. `PeerSet` is never shared, especially now that it is composed instead of derived. Only the surrounding types are shared, and they already have a lock for their internals that could be used to guard `PeerSet` just like it guards any other member. Then we could get rid of `PeerSet`'s lock.